garnett/trajectory.py

```diff
@@ -375,23 +359,79 @@ def __ne__(self, other):
                             details="Use to_hoomd_snapshot with no argument.")
     def make_snapshot(self):
         "Create a HOOMD-blue snapshot object from this frame."
-        self.load()
         return self.to_hoomd_snapshot()
 
     def to_hoomd_snapshot(self, snapshot=None):
         "Copy this frame to a HOOMD-blue snapshot."
         self.load()
-        return _to_hoomd_snapshot(self.frame_data, snapshot)
+        if snapshot is None:
+            try:
+                from hoomd import data
+            except ImportError:
+                try:
+                    # Try importing from hoomd 1.x
+                    from hoomd_script import data
+                except ImportError:
+                    raise ImportError('hoomd')
+            box = data.boxdim(
+                Lx=self.box.Lx,
+                Ly=self.box.Ly,
+                Lz=self.box.Lz,
+                xy=self.box.xy,
+                xz=self.box.xz,
+                yz=self.box.yz,
+                dimensions=self.box.dimensions,
+            )
+            snapshot = data.make_snapshot(
+                N=len(self),
+                box=box,
+                particle_types=self.types,
+            )
+            np.copyto(
+                snapshot.particles.typeid,
+                np.array(self.typeid, dtype=snapshot.particles.typeid.dtype)
+            )
+        for prop in PARTICLE_PROPERTIES:
+            try:
+                np.copyto(getattr(snapshot.particles, prop), getattr(self, prop))
+            except AttributeError:
+                pass
+        return snapshot
 
     @deprecation.deprecated(deprecated_in="0.7.0",
                             removed_in="0.8.0",
                             current_version=__version__,
                             details="Use to_hoomd_snapshot.")
     def copyto_snapshot(self, snapshot):
         "Copy this frame to a HOOMD-blue snapshot."
-        self.load()
```

Comment:
Same as above, why remove the `load` statement?

Reply:
This is called by `self.to_hoomd_snapshot` so it's redundant to have it twice.

Reply:
Ah, I missed that. Good then